docusaurus/docs/iOS/guides/working-with-messagelist-layout.md

```diff
@@ -0,0 +1,229 @@
+        var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+        options.remove([.flipped, .bubble, .timestamp, .avatar, .avatarSizePadding, .authorName, .threadInfo, .reactions])
+
+        // Little helper to figure out if the message is first in group.
+        let isFirstInGroup: Bool = {
+            let messageIndex = messages.index(messages.startIndex, offsetBy: indexPath.item)
+            let message = messages[messageIndex]
+            guard messageIndex < messages.index(before: messages.endIndex) else { return true }
+            let previousMessage = messages[messages.index(after: messageIndex)]
+            guard previousMessage.author == message.author else { return true }
+            let delay = previousMessage.createdAt.timeIntervalSince(message.createdAt)
+            return delay > minTimeIntervalBetweenMessagesInGroup
+        }()
+
+        // If the message is first in group and sent by someone, let's add metadata to it.
+        if isFirstInGroup {
+            options.insert([.avatar, .timestamp, .authorName])
+        } else {
+            // The message needs to have avatarView-sized padding otherwise the message below the first one would 
+            // have leading on the level of the avatar displayed in first message.
+            options.insert(.avatarSizePadding)
+        }
+        return options
+    }
+}
+```
+
+Now let's subclass the  `ChatMessageContentView`  and change the layout of it. 
+
+```swift 
+
+final class CustomChatMessageContentView: ChatMessageContentView {
+    override var maxContentWidthMultiplier: CGFloat { 1 }
+
+    // Let's override the layout function to implement custom layout:
+    override func layout(options: ChatMessageLayoutOptions) {
+        super.layout(options: options)
+
+        // To have the avatarView aligned at top with rest of the we need to set leading alignment to `mainContainer`.
+        mainContainer.alignment = .leading
+        
+        // Get subviews of container holding `bubbleContentContainer` when we disabled `.bubble` option.
+        let subviews = bubbleThreadMetaContainer.subviews
+        // Remove the subviews.
+        bubbleThreadMetaContainer.removeAllArrangedSubviews()
+        // Simply add the subviews in reversed order
+        bubbleThreadMetaContainer.addArrangedSubviews(subviews.reversed())
+        // By default, there are directionalLayoutMargins with system value because of the bubble border option.
+        // We need to disable them to get cleaner 
+        bubbleContentContainer.directionalLayoutMargins = .zero
+    }
+}
+
+```
+
+Everything that is left is to assign those custom subclasses to `Components` :
+
+```swift
+Components.default.messageLayoutOptionsResolver = CustomMessageOptionsResolver()
+Components.default.messageContentView = CustomChatMessageContentView.self // Assign type, not instance.
+```
+
+![Custom Chat layout](../assets/messagelist-layout-custom-final.png) 
```

Comment:
This image looks huge, it needs to be smaller

Reply:
made this image smaller, others thanks to table look good...